robocop/checkers/duplications.py

```diff
@@ -0,0 +1,96 @@
+"""
+Duplications checkers
+"""
+from collections import defaultdict
+from robocop.checkers import VisitorChecker
+from robocop.rules import RuleSeverity
+
+
+def register(linter):
+    linter.register_checker(DuplicationsChecker(linter))
+    pass
+
+
+class DuplicationsChecker(VisitorChecker):
+    """ Checker for duplicated names. """
+    rules = {
+        "0801": (
+            "duplicated-test-case",
+            'Multiple test cases with name "%s" in suite',
+            RuleSeverity.ERROR
+        ),
+        "0802": (
+            "duplicated-keyword",
+            'Multiple keywords with name "%s" in file',
+            RuleSeverity.ERROR
+        ),
+        "0803": (
+            "duplicated-variable",
+            'Multiple variables with name "%s" in Variables section. Note that Robot Framework is case insensitive',
+            RuleSeverity.ERROR
+        ),
+        "0804": (
+            "duplicated-resource",
+            'Multiple resource imports with name "%s" in suite',
+            RuleSeverity.WARNING
+        ),
+        "0805": (
+            "duplicated-library",
+            'Multiple library imports with name "%s" and identical arguments in suite',
+            RuleSeverity.WARNING
+        )
+    }
+
+    def __init__(self, *args):
+        self.test_cases = defaultdict(list)
+        self.keywords = defaultdict(list)
+        self.variables = defaultdict(list)
+        self.resources = defaultdict(list)
+        self.libraries = defaultdict(list)
+        super().__init__(*args)
+
+    def visit_File(self, node):
+        self.test_cases = defaultdict(list)
```

Comment:
Do you need to assign it again? You did it in init before.

Reply:
I need to init it in init and Im resetting them again in visit_File because I want to have empty containers before every file scan. Otherwise It would compare names between files